docs/getting-started/auth-self-defence-checks.adoc

```diff
@@ -0,0 +1,148 @@
+= Self Defence Checks
+
+As part of the Auth SDK module, there is a self defence check feature.  This feature offers multiple checks that can be used to analyse the secuirty of the device.
```

Comment:
I think we should provide a use case here for why these checks are useful. I added some info on the inteneded usage of this feature in this ticket https://issues.jboss.org/browse/AEROGEAR-1980 to get across for the demo video. I think we should also portray that information here.